src/main/src/main/java/org/geoserver/catalog/WMSLayerInfo.java

```diff
@@ -54,4 +54,10 @@
     public Set<StyleInfo> getStyles();
 
     public StyleInfo getDefaultStyle();
+
+    public void reloadRemoteStyles();
+
+    public List<StyleInfo> getAllAvailableRemoteStyles();
+
+    public void setAllAvailableRemoteStyles(List<StyleInfo> allAvailableRemoteStyles);
```

Comment:
In the GeoServer configuration there should not be setters for collections, all collections should be "live collections", that is, you only have a getter, the collection returned is never null, and you can modify it directly.

I see that other recent changes ignored it, there are setters for remote styles, also, it's "preferred", not "preffered" (typo in method name).

Reply:
Current implementation persists the available list of StyleInfo inside the WMSLayerInfo, because the StyleInfo carries remote URL of style itself on host server, which is used in cascading legend requests and generating capability document.

But your point makes sense, I will remove the live collection outside the bean class and use it only when needed, e.g UI, generating capability document etc

Reply:
will remove the getter